packages/contentstack-launch/test/unit/commands/z-functions.test.ts

```diff
@@ -0,0 +1,54 @@
+import { describe, it } from 'mocha';
+import { expect } from 'chai';
+import { stub, spy, assert } from 'sinon';
+import Functions from '../../../src/commands/launch/functions';
+import { FunctionsDirectoryNotFoundError } from '../../../src/util/cloud-function/errors/cloud-function.errors';
+import fs from 'fs';
+import express from 'express';
+
+describe('Functions Command', () => {
+  const doesFunctionsFolderExist = fs.existsSync('./functions');
+  it('should throw an error if functions folder does not exist in root directory or should log folder exists', async () => {
+    const errorFunction = spy(() => {
+      throw new FunctionsDirectoryNotFoundError(process.cwd());
+    });
+    const successFunction = () => console.log('Functions folder exists');
+
+    if (doesFunctionsFolderExist === false) {
+      expect(errorFunction).to.throw(Error, `No functions directory found at '${process.cwd()}'`);
+    } else {
+      successFunction();
+    }
+  });
+
+  it('Should show "No Serverless functions detected" when no files are there in functions folder or "Detected Serverless functions" if files are detected', async () => {
+    if (doesFunctionsFolderExist === false) {
+      fs.mkdirSync('./functions');
+    }
+    const app = express();
+    const processStub = stub(process, 'exit');
+    const logStub = stub(console, 'log');
+    const doFilesExistInFolder = fs.readdirSync('./functions');
+    await Functions.run(['-p', '4050']);
+    if (doFilesExistInFolder.length === 0) {
+      assert.calledOnce(processStub);
+      fs.writeFileSync(
+        './functions/user.js',
+        `export default function user(req, res) {
+            res.json({ data: 'Welcome' });
+          }
+          `,
+      );
+    } else {
+      assert.notCalled(processStub);
+      assert.calledWithExactly(logStub, 'Detected Serverless functions...');
+      assert.calledWithExactly(logStub, `Serving on port 4050`);
+    }
+    processStub.restore();
+    logStub.restore();
+    app.listen().close((err) => {
+      console.log('closing server');
+      process.exit(err ? 1 : 0);
+    });
+  });
+});
```

Comment:
@akashbhosale97 Is this file pushed intentionally or by mistake? I could see two files for the functions command.

Reply:
it is intentional,  there is only one file just the file name was changed

Reply:
Any specific reason to have the filename prefixed with `z-` .?

Reply:
the functions command has express server which keeps on running and wont close, if close then and there then next test commands wont run so changed it so that all test cases run and this one to run at last and then server can be closed